### PR TITLE
get the dual solution also for singleton constraints

### PR DESCRIPTION
**Pull request by mattmilten**

### CODE REVIEW

**Comment by mattmilten** [comment]
fixes #92

File: src/pyscipopt/scip.pyx [hunk 1]
```diff
@@ -1276,9 +1276,8 @@ cdef class Model:
                 transcons = <Constraint>self.getTransformedCons(cons)
             else:
                 transcons = cons
-            if _nvars > 1:
-                dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
-            else:
+            dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
+            if dual == 0.0 and _nvars == 1:
```
Comment (fserra) [inline src/pyscipopt/scip.pyx]: I don't quite get the change. Looks very cosmetic to me. Also, why would we trust `SCIPgetDualsolLinear` if it returns nonzero when the constraint contains only one variable?

File: tests/test_pricer.py [hunk 2]
```diff
@@ -132,15 +132,15 @@ def test_cuttingstock():
...
+            outline = 'Pattern_' + str(i) + ':\t' + str(solValue) + '\t\tCuts:\t '
             for j in range(len(widths)):
                 rollUsage += pricer.data['patterns'][i][j]*widths[j]
                 widthOutput[j] += pricer.data['patterns'][i][j]*solValue
                 outline += str(pricer.data['patterns'][i][j]) + '\t'
             outline += 'Usage:' + str(rollUsage)
             print(outline)
 
-    print('\t\t\tTotal Output:','\t'.join(str(e) for e in widthOutput))
+    print('\t\t\tTotal Output:\t', '\t'.join(str(e) for e in widthOutput))
```
Comment (fserra) [inline tests/test_pricer.py]: The test is still not testing anything. Could you add some asserts of what is expected?

Comment (mattmilten) [reply tests/test_pricer.py]: Yes, good idea! Will do that.

File: src/pyscipopt/scip.pyx (hunk 1 above)
Comment (mattmilten) [reply src/pyscipopt/scip.pyx]: Apparently, sometimes a bound constraint appears as a constraint in the LP and we actually get a proper dual value while at other times, it gets converted into variable bounds and we need to ask for the reduced costs. In this case, the `getDualsolLinear()` call returns 0. Yeah, it is a workaround, but until we don't properly implement this in SCIP itself, we have to deal with it somehow.

**Pull request merged**